### DISCUSSION TITLE
A custom backend for PocketBase

### DISCUSSION THREAD

**Discussion by hariso** [cschleiden/go-workflows#235] [Ideas]
Hello!

I'm exploring the idea of writing a PocketBase backend (which uses SQLite). Since this is quite specific and every backend also requires maintenance, I understand that it shouldn't be part of this project, and should be an "external" backend. 

go-workflows has an SQLite backend, but it needs a path to the DB file to create it. That's a little tricky because of the write locks for SQLite.

I've considered a couple of other options, but it looks like the best one in the end is a PocketBase backend. However, the [`backend.Backend`](https://github.com/cschleiden/go-workflows/blob/main/backend/backend.go#L27) interface uses a few types which are internal.

So what do you think about making all the types which `backend.Backend` uses "public"? Or is there maybe another option?

**Comment by cschleiden** [comment]
No you're right, I think that's the way to go. I haven't checked this, if there a non exported types they should be. Filed https://github.com/cschleiden/go-workflows/issues/238 to track this.

**Reply by hariso** [reply]
Thanks, Christopher!

**Reply by cschleiden** [reply]
This is implemented now.

**Reply by hariso** [reply]
Awesome, thanks a lot!
